crates/ruff/src/checkers/ast/traits.rs

```diff
@@ -0,0 +1,42 @@
+use ruff_diagnostics::Diagnostic;
+
+use crate::checkers::ast::RuleContext;
+use crate::registry::Rule;
+use crate::settings::Settings;
+
+/// Trait for a lint rule that can be run on an AST node of type `T`.
+pub(crate) trait Analyzer<T>: Sized {
+    /// The [`Rule`] that this analyzer implements.
+    fn rule() -> Rule;
```

Comment:
Including `Rule` as part of the trait lets us implement `enabled` below on `RegisteredRule`.

Reply:
I don't mind this but I think I would favor renaming `Rule` to e.g. `RuleKind` and name this trait `Rule` instead (it's this trait that implements the logic). Ideally, we would move more metadata to this rule to avoid having `Violation`, `RuleKind` and `Rule`